test/executable/execute.ts

```diff
@@ -0,0 +1,644 @@
+            alreadyCheckedFileManager = true;
+          }
+        }
+      } catch (readError) {
+        console.error(
+          `[${connectorName}] 파일 읽기 오류 ${filePath}:`,
+          readError,
+        );
+      }
+    } // end of for loop
+
+    // 4. 결과 반환 전 최종 로그
+    if (!foundEnvList) {
+      console.log(`[${connectorName}] ENV_LIST not found in ${structuresDir}`);
+    }
+    if (!alreadyCheckedFileManager) {
+      // IProps 정의 자체를 못 찾은 경우
+      console.log(
+        `[${connectorName}] IProps definition not found in any file within ${structuresDir}`,
+      );
+    } else if (!hasFileManager && !connectorName.includes("s3")) {
+      // IProps는 찾았지만 FileManager가 없는 경우 (s3 제외)
+      console.log(
+        `[${connectorName}] FileManager type was not found within the IProps definition.`,
+      );
+    }
+
+    return { envList, hasFileManager };
+  } catch (error) {
+    console.error(`[${connectorName}] 서비스 정보 추출 중 오류 발생:`, error);
+    return { envList: [], hasFileManager: false };
+  }
+}
+
+// --- 이전에 Service 파일 생성자를 확인하던 로직은 제거됨 ---
+
+// Helper function to convert SNAKE_CASE to camelCase
+function toCamelCase(str: string): string {
+  return str.toLowerCase().replace(/_([a-z])/g, (_, p1) => p1.toUpperCase());
+}
+
+async function updateServiceFileWithEnvVars(
+  projectDir: string,
+  envVars: string[],
+  hasFileManager: boolean,
+): Promise<void> {
+  const indexPath = path.join(projectDir, "src", "index.ts");
+
+  try {
+    if (!fs.existsSync(indexPath)) {
+      console.error(`❌ index.ts 파일을 찾을 수 없습니다: ${indexPath}`);
+      return;
+    }
+
+    let content = fs.readFileSync(indexPath, "utf8");
+    let serviceName = ""; // 찾은 서비스 클래스 이름 저장
+
+    // 정규 표현식 수정: execute: new ServiceName(...) 부분을 찾고 ServiceName을 캡처
+    // 생성자 인자가 없거나, 하나 이상 있는 경우 모두 포함
+    const constructorRegex =
+      /(\bexecute\s*:\s*new\s+)(\w+Service)(\s*\([^)]*\))/g; // global flag 유지
+    const match = constructorRegex.exec(content); // 첫 번째 매치 찾기
+
+    if (!match) {
+      console.warn(
+        `⚠️ ${indexPath} 파일에서 'execute: new ...Service(...)' 패턴을 찾을 수 없습니다.`,
+      );
+      return; // 일치하는 패턴 없음
+    }
+
+    serviceName = match[2]; // 매치된 그룹에서 서비스 이름 추출 (예: "StableDiffusionBetaService")
+    const originalConstructorCall = match[0]; // 원본 생성자 호출 부분 저장 (예: "execute: new StableDiffusionBetaService()")
+    console.log(`ℹ️ 대상 서비스 감지: ${serviceName}`);
+    console.log(`ℹ️ 원본 생성자 호출: ${originalConstructorCall}`);
+
+    // --- AWS S3 Service 처리 (필요한 경우) ---
+    let awsPropsStr = "";
+    if (hasFileManager) {
+      const awsEnvVars = [
+        "AWS_ACCESS_KEY_ID",
+        "AWS_SECRET_ACCESS_KEY",
+        "AWS_S3_BUCKET",
+        "AWS_S3_REGION",
+      ];
+      awsPropsStr = "{\n"; // 여기서 awsPropsStr 초기화
+      awsEnvVars.forEach((envVar) => {
+        const camelCaseKey = toCamelCase(envVar);
+        awsPropsStr += `        ${camelCaseKey}: process.env.${envVar}!,\n`;
+      });
+      awsPropsStr += "      }"; // 들여쓰기 조정
```

Comment:
In the future, new FileManager can be created like Azure, GCP, LocalFileSystem, and so on.
so this part will be changed someday.